File: randomGen.cpp
#include "randomGen.h"
#include <climits>
#include <iostream>

randomGen::randomGen(int seed){
    this->seed = seed;
    modulo = UINT_MAX;
    constant =  1664525;
    multiplier = 1013904223;
    lastRandom = seed;
}

int randomGen::getSeed(){
    return seed;
}

void randomGen::setSeed(int newSeed){
    lastRandom = newSeed;
    seed = newSeed;
}

double randomGen::getRandom(double lowerLimit, double upperLimit){
    unsigned int randomNumber = ((multiplier*lastRandom)+constant) % modulo;
    lastRandom = randomNumber;
    //Now convert inbetween limits
    double randomDouble = ((((double) randomNumber)/UINT_MAX)*(upperLimit-lowerLimit))+lowerLimit;
    return randomDouble;
}

File: randomGen.py
class randomGen:
    
    seed = 0
    lastRandom = 0
    multiplier =  0
    constant = 0
    modulo = 0

    def __init__(self,seed):
        self.seed = seed
        self.modulo = 2**32
        self.constant =  1664525
        self.multiplier = 1013904223
        self.lastRandom = seed

    def getSeed(self):
        return self.seed

    def setSeed(self,seed):
        self.seed = seed

    
    def getRandom(self,lowerLimit,upperLimit):
        randomNumber = ((self.multiplier*self.lastRandom)+self.constant) % self.modulo
        self.lastRandom = randomNumber
        randomFloat = ((float(randomNumber)/self.modulo)*(upperLimit-lowerLimit))+lowerLimit
        return randomFloat


